### PR TITLE
TandemFoil: T_max sweep (5/15) and gc=0.5 at winning lr=1.25e-4

### PR DESCRIPTION
**Pull request by tcapelle**
## Hypothesis

T_max=10 is the current TandemFoil baseline (PR #2810, 30.10). The AirfRANS breakthrough (PR #2823) showed T_max=10 dramatically outperforms T_max=5 for that dataset. But TandemFoil is a structurally different problem — it uses Lion optimizer (not AdamW), has physics-informed features, and its gradient landscape may respond differently to cosine cycle length. We test T_max=5 (shorter cycles) and T_max=15 (longer cycles) at the winning lr=1.25e-4 to check if TandemFoil has a different optimal cycle length. We also test gc=0.5 + T_max=5 as a compound since gc=0.5 was never tested at T_max values other than the baseline.

## Baseline

- **Primary metric:** `val_primary/surface_pressure_mae` (lower is better)
- **Current best:** 30.10 (PR #2810, 3L/192d + Lion lr=1.25e-4 + gc=1.0 + T_max=10 + WD=1e-2 + slices=64)

## Runs

Run all 3 simultaneously (1 GPU each). Report **best val_primary/surface_pressure_mae checkpoint** (NOT final epoch).

**Run 1 (CUDA 0): T_max=5 + gc=1.0**
```bash
cd target/icml2026 && CUDA_VISIBLE_DEVICES=0 SENPAI_MAX_EPOCHS=9999 SENPAI_TIMEOUT_MINUTES=180 python train.py \
  --dataset tandemfoil --optimizer lion --lr 1.25e-4 \
  --cosine-t-max 5 --grad-clip 1.0 --weight-decay 1e-2 \
  --no-use-ema --model-slices 64 \
  --model-layers 3 --model-hidden-dim 192 --model-heads 3 \
  --enable-fourier --enable-te-coord-frame --enable-cp-panel \
  --enable-cp-panel-tandem-only --asinh-pressure \
  --residual-prediction --enable-pressure-prior-addition \
  --epochs 999 \
  --wandb-group tandem-Tmax-sweep \
  --wandb-name tandem-lr125e4-gc10-Tmax5
```

**Run 2 (CUDA 1): T_max=5 + gc=0.5**
```bash
cd target/icml2026 && CUDA_VISIBLE_DEVICES=1 SENPAI_MAX_EPOCHS=9999 SENPAI_TIMEOUT_MINUTES=180 python train.py \
  --dataset tandemfoil --optimizer lion --lr 1.25e-4 \
  --cosine-t-max 5 --grad-clip 0.5 --weight-decay 1e-2 \
  --no-use-ema --model-slices 64 \
  --model-layers 3 --model-hidden-dim 192 --model-heads 3 \
  --enable-fourier --enable-te-coord-frame --enable-cp-panel \
  --enable-cp-panel-tandem-only --asinh-pressure \
  --residual-prediction --enable-pressure-prior-addition \
  --epochs 999 \
  --wandb-group tandem-Tmax-sweep \
  --wandb-name tandem-lr125e4-gc05-Tmax5
```

**Run 3 (CUDA 2): T_max=15 + gc=1.0**
```bash
cd target/icml2026 && CUDA_VISIBLE_DEVICES=2 SENPAI_MAX_EPOCHS=9999 SENPAI_TIMEOUT_MINUTES=180 python train.py \
  --dataset tandemfoil --optimizer lion --lr 1.25e-4 \
  --cosine-t-max 15 --grad-clip 1.0 --weight-decay 1e-2 \
  --no-use-ema --model-slices 64 \
  --model-layers 3 --model-hidden-dim 192 --model-heads 3 \
  --enable-fourier --enable-te-coord-frame --enable-cp-panel \
  --enable-cp-panel-tandem-only --asinh-pressure \
  --residual-prediction --enable-pressure-prior-addition \
  --epochs 999 \
  --wandb-group tandem-Tmax-sweep \
  --wandb-name tandem-lr125e4-gc10-Tmax15
```

## Expected Outcome

If TandemFoil follows AirfRANS, T_max=15 may beat T_max=10. Alternatively, TandemFoil's Lion optimizer and physics features may prefer shorter cycles. We expect the T_max=5 gc=1.0 run to likely be worse (fewer cosine minima per unit time), but gc=0.5 + T_max=5 tests whether tighter clipping compensates. Any result below 30.10 is a win.

Report: best val_primary/surface_pressure_mae for each run, epoch of best, and gradient norm behavior.

### CODE REVIEW

**Comment by morganmcg1** [comment]
Closing this PR because the assigned student pod was retired during the DrivAerML and cross-dataset refocus on 2026-04-21. We are freeing capacity for the new relaunch wave, and this lane is no longer being actively pursued. Any useful results already logged here or in W&B remain part of the research record.
